src/jdk.compiler/share/classes/com/sun/tools/javac/launcher/Main.java

```diff
@@ -407,37 +410,70 @@ private String compile(Path file, List<String> javacOpts, Context context) throw
         return mainClassName;
     }
 
+    private static boolean isPublic(Method method) {
+        return method != null && Modifier.isPublic(method.getModifiers());
+    }
+
+    private static boolean isStatic(Method method) {
+        return method != null && Modifier.isStatic(method.getModifiers());
+    }
+
     /**
      * Invokes the {@code main} method of a specified class, using a class loader that
      * will load recently compiled classes from memory.
      *
      * @param mainClassName the class to be executed
-     * @param appArgs the arguments for the {@code main} method
+     * @param mainArgs the arguments for the {@code main} method
      * @param context the context for the class to be executed
      * @throws Fault if there is a problem finding or invoking the {@code main} method
      * @throws InvocationTargetException if the {@code main} method throws an exception
      */
-    private void execute(String mainClassName, String[] appArgs, Context context)
+    private void execute(String mainClassName, String[] mainArgs, Context context)
             throws Fault, InvocationTargetException {
         System.setProperty("jdk.launcher.sourcefile", context.file.toString());
         ClassLoader cl = context.getClassLoader(ClassLoader.getSystemClassLoader());
         try {
             Class<?> appClass = Class.forName(mainClassName, true, cl);
-            Method main = appClass.getDeclaredMethod("main", String[].class);
-            int PUBLIC_STATIC = Modifier.PUBLIC | Modifier.STATIC;
-            if ((main.getModifiers() & PUBLIC_STATIC) != PUBLIC_STATIC) {
+            Method mainMethod = MainMethodFinder.findMainMethod(appClass);
+            int mods = mainMethod.getModifiers();
+            boolean isStatic = Modifier.isStatic(mods);
+            boolean isPublic = Modifier.isStatic(mods);
```

Comment:
This looks like a typo. This should have been `Modifier.isPublic(mods);`

Reply:
Changing